ompi/mca/osc/rdma/osc_rdma_peer.c

```diff
@@ -73,7 +73,19 @@ static int ompi_osc_rdma_peer_btl_endpoint (struct ompi_osc_rdma_module_t *modul
         }
     }
 
-    /* unlikely but can happen when creating a peer for self */
+    if (peer_id == ompi_comm_rank (module->comm)) {
```

Comment:
I'm really concerned about this approach.  I suppose it works, but it feels like the right place for this decision is during the initial btl selection logic.